clients/filesystem-hadoop3/src/main/java/org/apache/gravitino/filesystem/hadoop/DefaultGravitinoFileSystemCredentialsProvider.java

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.filesystem.hadoop;
+
+import com.google.common.base.Preconditions;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.catalog.hadoop.fs.GravitinoFileSystemCredentialsProvider;
+import org.apache.gravitino.client.GravitinoClient;
+import org.apache.gravitino.credential.Credential;
+import org.apache.gravitino.file.Fileset;
+import org.apache.gravitino.file.FilesetCatalog;
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * Default implementation of {@link GravitinoFileSystemCredentialsProvider} which provides
+ * credentials for Gravitino Virtual File System.
+ */
+public class DefaultGravitinoFileSystemCredentialsProvider
+    implements GravitinoFileSystemCredentialsProvider {
+
+  private Configuration configuration;
+
+  private static final Pattern IDENTIFIER_PATTERN =
+      Pattern.compile("^(?:gvfs://fileset)?/([^/]+)/([^/]+)/([^/]+)(?>/[^/]+)*/?$");
+
+  @Override
+  public void setConf(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  @Override
+  public Configuration getConf() {
+    return configuration;
+  }
+
+  @Override
+  public Credential[] getCredentials() {
+    String virtualPath = configuration.get(GVFS_PATH);
```

Comment:
How about passing fileset NameIdentifier directly?

Reply:
@jerryshao believe using `NameIdentifier ` direcctly is hacky, so is there a better solution than this two

Reply:
I never said using `NameIdentifier` is bad and using this is good. Don't misinterpret my meaning. What I said is using static configuration to pass the dynamic variable is bad.

Reply:
@yuqi1129 could you change to `NameIdentifier`? I couldn't see the benefit to use path